### PR TITLE
feat(auth): Make token auth default

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- make token auth the default being returned from authentication endpoints in api-v2
- Add `auth/session` to convert token to session based auth
- add regex-scopes to authenticate middleware

### CODE REVIEW

File: packages/medusa/src/utils/authenticate-middleware.ts [hunk 1]
```diff
@@ -14,8 +15,15 @@ type MedusaSession = {
...
+const isValidScope = (authScope: string | RegExp, scopeToVerify: string) => {
+  if (authScope instanceof RegExp) {
+    return authScope.test(scopeToVerify)
+  }
+  return scopeToVerify === authScope
+}
+
 export const authenticate = (
-  authScope: string,
+  authScope: string | RegExp,
```
Comment (srindom) [inline packages/medusa/src/utils/authenticate-middleware.ts:19]: nice trick 👍

File: packages/medusa/src/utils/authenticate-middleware.ts [hunk 2]
```diff
@@ -25,34 +33,41 @@ export const authenticate = (
...
-              .retrieveAuthUserFromJwtToken(token, authScope)
-              .catch(() => null)
+          if (tokenType.toLowerCase() === BEARER_AUTH) {
+            // get config jwt secret
+            // verify token and set authUser
+            const { jwt_secret } =
+              req.scope.resolve("configModule").projectConfig
+
+            const verified = jwt.verify(token, jwt_secret) as JwtPayload
+
```
Comment (srindom) [inline packages/medusa/src/utils/authenticate-middleware.ts:61]: thought: maybe this should move to a util.

**Review by srindom — APPROVED** [review]
LGTM - needs to rebase on develop and fix tests.

**Review by adrien2p — APPROVED** [review]

File: packages/medusa/src/utils/authenticate-middleware.ts (hunk 2 above)
Comment (pKorsholm) [reply packages/medusa/src/utils/authenticate-middleware.ts:61]: Added `stringEqualsOrRegexMatch` util 😄

Comment (pKorsholm) [reply packages/medusa/src/utils/authenticate-middleware.ts:61]: Added a util for string equality/regex matching 😄

**Comment by srindom** [comment]
@pKorsholm - let's merge this and get #6314 in after 👍

**Pull request merged**